### PR TITLE
CTCTRADERS-141:check your answers navigation and submission

### PR DESCRIPTION
**Pull request by ashwinirudrappa**

### CODE REVIEW

File: app/controllers/CheckEventAnswersController.scala [hunk 1]
```diff
@@ -43,16 +44,24 @@ class CheckEventAnswersController @Inject()(
...
+      implicit request =>
+      Future.successful(Redirect(routes.CheckYourAnswersController.onPageLoad(mrn)))
+    }
+
+  private def completeSections(userAnswers: UserAnswers)(implicit messages: Messages): Seq[Section] = {
     val helper = new CheckYourAnswersHelper(userAnswers)
+    Seq(Section(None, eventsSection(helper)))
+  }
+
+  private def eventsSection(helper: CheckYourAnswersHelper): Seq[Row] = {
```
Comment (garethgooch) [inline app/controllers/CheckEventAnswersController.scala]: Do we need the val events here? wouldn't this be simpler as def eventsSection = Seq(...,..).flatten

File: app/services/conversion/ArrivalNotificationConversionService.scala [hunk 2]
```diff
@@ -36,26 +36,54 @@ class ArrivalNotificationConversionService {
...
+        notificationDate = LocalDate.now(),
+        customsSubPlace = Some(customsSubPlace),
+        trader = traderAddress(tradersAddress, traderEori, traderName),
+        presentationOffice = presentationOffice,
+        enRouteEvents = enRouteEvents(userAnswers)
       )
     }
   }
 
+  private def eventDetails(isTranshipment: Boolean, userAnswers: UserAnswers): EventDetails = {
```
Comment (garethgooch) [inline app/services/conversion/ArrivalNotificationConversionService.scala]: We are passing in UserAnswers, but where we call this we are getting isTranshipment from UserAnswers, wouldn't this be better to either pull everything out and pass it in, or have the def do all the leg work?

File: app/controllers/CheckEventAnswersController.scala (hunk 1 above)
Comment (ashwinirudrappa) [reply app/controllers/CheckEventAnswersController.scala]: @garethgooch we need helper to get sections that's the reason its def not val

File: test/services/conversion/ArrivalNotificationConversionServiceSpec.scala [hunk 3]
```diff
@@ -38,48 +39,88 @@ class ArrivalNotificationConversionServiceSpec extends SpecBase with ScalaCheckP
...
+    }
+
+    "must return 'Normal Arrival Notification' message when there is on incident on route" in {
+      forAll(arbitrary[NormalNotification], generatorTraderWithEoriAllValues, Gen.alphaNumStr, arbitrary[EnRouteEvent]) {
+        case (arbArrivalNotification, trader, subPlace, enRouteEvent) =>
+
+          val incidentInformation = enRouteEvent.eventDetails match {
+            case incident:Incident => incident.information
+            case _ => None
+          }
```
Comment (amaalali) [inline test/services/conversion/ArrivalNotificationConversionServiceSpec.scala]: We shouldn't have this logic in tests, we can control the test data that we are using by writing custom generators for this test

**Pull request merged**